### PR TITLE
fix(DB): Rewrite Supplies to Auberdine quest

### LINKED ISSUE #6843: [Ashenvale] Quest Supplies to Auberdine - First Ambush Ignored by Feero Ironhand
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [ ] 1-19
- [x] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

Original report: [chromiecraft/issues/#1127](https://github.com/chromiecraft/chromiecraft/issues/1127**)

##### CURRENT BEHAVIOUR:

Quest: Supplies to Auberdine (https://classic.wowhead.com/quest=976/supplies-to-auberdine)

In short, I have to escort an NPC (Feero Ironhand https://classic.wowhead.com/npc=4484/feero-ironhand). During the escort, there are supposed to be 3 ambushes. During the first ambush, the NPC (Feero Ironhand) completely ignored the mobs as I aggroed them. This does not happen in the second and third ambush (it works as expected, Feero Ironhand helped me hit the mobs).

##### EXPECTED BLIZZLIKE BEHAVIOUR:

The NPC Feero Ironhand should not ignore the mobs during the first ambush.

##### SOURCE:

I can't reproduce the bug again since I have turned the quest in, but the expected blizzlike behaviour should be like this: https://www.youtube.com/watch?v=xIaSP8YE_xg

##### STEPS TO REPRODUCE THE PROBLEM:

1. Accept quest Supplies to Auberdine
2. Follow NPC Feero Ironhand until the first ambush
3. NPC Feero Ironhand will ignore the mobs during the first ambush
4.

##### EXTRA NOTES:

Edit: I am not really sure, but I might have been invisible as a druid in cat form during the first ambush. Hope this info can help

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/f16fe3f8dc5e6fd0af8848714fab2195cb1182c1

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

There was a similar issue alsmost three years ago, but even after the fix, the quest still has some problems: [chromiecraft/issues/#395](https://github.com/chromiecraft/chromiecraft/issues/395)

Full credit to @adriansutanahadi for finding out, how to reproduce the bug and helping with further tests, he wrote:
I have been able to reproduce it. Apparently, I was on bear form, not prowling on cat form.

Steps to reproduce:

1. Accept quest Supplies to Auberdine
2. Follow NPC Feero Ironhand until the first ambush
3. Taunt the mobs before NPC Feero Ironhand have the chance of being hit
4. NPC Feero Ironhand will ignore the mobs during the first ambush

grafik

I haven't checked it with the second and third ambush though.

Edit: core version was rev. 2f1be74d5b82+ 2021-07-08 16:03:06 +0000 (master branch)

@Efymer also added:

This quest has some issues:

TBC video: https://www.youtube.com/watch?v=NZaDRm3E9T4

- The NPC Feero Ironhand should be running instead of walking.
- Spawn points of Assassins is wrong.
- First assasins wave is using caster spells being melee and not attacking the NPC Feero Ironhand (I could reproduce it on my AC env).

And I just checked until the first wave...

### PR DESCRIPTION
**Pull request by Efymer**
## Changes Proposed:
https://www.youtube.com/watch?v=NZaDRm3E9T4

All the changes made in this PR are based on this video from Classic TBC to adjust as much possible the Blizzlike feeling of the quest which had many issues:

- Quest should reward the whole group and not just the stored target.
- The escorted NPC should run instead of walk.
- Waypoints were wrong, the quest was supposed to end in Darkshore.
- First wave spawns were wrong, creatures are supposed to ambush from outside the path.
- First wave spawns had wrong spells.
- Second wave spawns were wrong, creatures are supposed to ambush from outside the path.
- Second wave spawns had wrong spells.
- Third wave had a totally different waypoint of what was supposed to be.
- Third wave spawns had wrong spells and missing texts.

## Issues Addressed:

- Closes #6843
- Closes https://github.com/chromiecraft/chromiecraft/issues/1127

## SOURCE:

- https://www.youtube.com/watch?v=NZaDRm3E9T4

## Tests Performed:

- Tested multiple times in my local AC

## How to Test the Changes:

1.  Test the quest with this video next to it: https://www.youtube.com/watch?v=NZaDRm3E9T4
2. Complete previous quest https://classic.wowhead.com/quest=973/the-tower-of-althalaxx
3. `.go c id 4484`
4. Start the escort quest.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
https://www.azerothcore.org/wiki/sql-standards

File: data/sql/updates/pending_db_world/rev_1625825866239168500.sql [hunk 1]
```diff
@@ -0,0 +1,158 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1625825866239168500');
+
+DELETE FROM `creature_summon_groups` WHERE summonerId = 4484;
```
Comment (LukasVolgger) [inline data/sql/updates/pending_db_world/rev_1625825866239168500.sql:3]: Please add backticks ` around keywords

**Review by LukasVolgger — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1625825866239168500.sql [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+-- Actionlist 2
+DELETE FROM `smart_scripts` WHERE (`source_type` = 9 AND `entryorguid` = 448402);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(448402, 9, 0, 0, 0, 0, 100, 0, 2000, 2000, 0, 0, 0, 1, 4, 5000, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Feero Ironhand - Actionlist - Say Line 4'),
+(448402, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 107, 2, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Feero Ironhand - Actionlist - Summon Creature Group 2'),
+(448402, 9, 2, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 25, 10, 0, 0, 0, 0, 0, 0, 0, 'Feero Ironhand - Actionlist - Start Attacking');
+
+
+DELETE FROM `waypoints` WHERE entry = 4484;
```
Comment (LukasVolgger) [inline data/sql/updates/pending_db_world/rev_1625825866239168500.sql:111]: Backticks

**Review by LukasVolgger — CHANGES_REQUESTED** [review]

**Review by LukasVolgger — APPROVED** [review]

**Comment by temperrr** [comment]
Tested.

- [x] Prerequisite quest required before you can take the quest
- [x] Upon completion both party members received credit
- [x] Encounters spawned
- [x] Last encounter used abilities correctly
- [ ]After each encounter, he should stand still, catching his breath, say his voice line above his head, and after that is done he should continue walking. Right now he immediately rushes to the next encounter. For the first encounter the time between chat gossip and movement should be  about 11 seconds. For the second encounter this break is a little shorter, about 7/8 seconds. Deriving these numbers from the video you linked

**Comment by FrancescoBorzi** [comment]
good job @Efymer and thanks @LukasVolgger and @temperrr for reviewing/testing this one

**Pull request merged**